compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1725,7 +1725,8 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
             AggregateKind::Adt(adt_did, variant_index, substs, _, active_field_index) => {
                 let def = tcx.adt_def(adt_did);
                 let variant = &def.variant(variant_index);
-                let adj_field_index = active_field_index.unwrap_or(field_index);
+                let adj_field_index =
+                    FieldIdx::from_usize(active_field_index.unwrap_or(field_index));
```

Comment:
This is the kind of place that will hopefully be temporary -- changing `AggregateKind::Adt` to hold a `FieldIdx` and the function to use `field_index: FieldIdx` instead of `field_index: usize` should get rid of this in an upcoming PR in this series, but I needed to stop the tendrils somewhere 🙃